qiskit_ibm_runtime/job_status.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""This module defines an enumerated type for the state of primitive jobs"""
+
+import enum
+
+
+class JobStatus(enum.Enum):
```

Comment:
If you use an enum for job status, and we introduce a new status, wouldn't that break downlevel providers?

Reply:
Should strings be returned instead? As mentioned in https://github.com/Qiskit/qiskit/issues/11541